server/cmwell-ws/app/logic/services/ServicesRoutesCache.scala

```diff
@@ -0,0 +1,94 @@
+/**
+  * Copyright 2015 Thomson Reuters
+  *
+  * Licensed under the Apache License, Version 2.0 (the “License”); you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *   http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  * an “AS IS” BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package logic.services
+
+import akka.actor.{Actor, ActorSystem, Props}
+import cmwell.domain.Infoton
+import cmwell.fts.PathFilter
+import cmwell.ws.Settings
+import com.typesafe.scalalogging.LazyLogging
+import javax.inject.{Inject, Singleton}
+import logic.CRUDServiceFS
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+import akka.pattern.ask
+import akka.util.Timeout
+import scala.concurrent.duration.DurationInt
+
+@Singleton
+class ServicesRoutesCache @Inject()(crudService: CRUDServiceFS)(implicit ec: ExecutionContext, sys: ActorSystem) extends LazyLogging {
+  private var services: Map[String, ServiceDefinition] = Map.empty
+  private val helperActor = sys.actorOf(Props(new ServicesRoutesCacheHelperActor), "ServicesRoutesCacheHelperActor")
+  private implicit val timeout: Timeout = 60.seconds
+
+  def find(path: String): Option[ServiceDefinition] =
+    services.find { case (route, _) => path.startsWith(route) }.map(_._2)
+
+  def list: Set[String] = services.keySet
```

Comment:
I think servicesList will be a better name

Reply:
but then the usag would be `servicesRoutesCache.servicesList` - we already know those are services, so .list feels like a better fit IMO.